nautobot/extras/api/views.py

```diff
@@ -530,21 +531,73 @@ def _run_job(request, job_model, legacy_response=False):
         raise MethodNotAllowed(request.method, detail="This job's source code could not be located and cannot be run")
     job = job_class()
 
-    input_serializer = serializers.JobInputSerializer(data=request.data, context={"request": request})
-    input_serializer.is_valid(raise_exception=True)
+    # default to first queue in job_model.task_queues if task_queue not specified
+    valid_queues = job_model.task_queues if job_model.task_queues else [settings.CELERY_TASK_DEFAULT_QUEUE]
+
+    # We need to call request.data for both cases as this is what pulls and caches the request data
+    data = request.data
+    files = None
+    schedule_data = None
+
+    # We must extract from the request:
+    # - Job Form data (for submission to the job itself)
+    # - Schedule data
+    # - Commit flag state
+    # - Desired task queue
+    # Depending on request content type (largely for backwards compatibility) the keys at which these are found are different
+    if "multipart/form-data" in request.content_type:
+        data = request._data.dict()  # .data will return data and files, we just want the data
+        files = request.FILES
+
+        # JobMultiPartInputSerializer is a "flattened" version of JobInputSerializer
+        input_serializer = serializers.JobMultiPartInputSerializer(data=data, context={"request": request})
+        input_serializer.is_valid(raise_exception=True)
+
+        commit = input_serializer.validated_data.get("_commit", None)
+        task_queue = input_serializer.validated_data.get("_task_queue", valid_queues[0])
```

Comment:
This is one of two places where `valid_queues[0]` is referenced after `valid_queues` is defined. Can we please give this its own variable to increase readability and DRY?